### ISSUE TITLE
Bashdatacatalog list contains unnecessary files

### ISSUE THREAD

**Issue by 1Dandan** [LiamBindle/bashdatacatalog#4]
# Problem description
Hi Liam, I am using `bashdatacatalog` to download missing files. It is a great tool but I do have some questions:
1) When there is existing entries for one year, it will fetch all missing files in that entry even though it is not in the time range specified. For example, I have downloaded some offline dust files for one week in 2019, and `bashdatacatalog-fetch` all files in the whole year 2019 although I have specified I time range is `2017-12-01,2018-12-31` using `bashdatacatalog-list`.

2) Also, I thought by specifying the meteorology is `GEOS-FP` in `MeteorologicalInputs.csv` , it will only fetch fine resolution such as 0.25 degree OFFLINE_DUST instead of 0.5 degree. Now it will also download 0.5 degree files

3) Another problem is it will download a lot of files for vary old years, which currently I am not sure why. For example, for the emissions of `SfcVMR` in `HEMCO_Config.rc`: `    --> SfcVMR                 :       true     # 1750-2014`,  I think it only needs files for certain years? But it will download all years' files including `./HEMCO/SfcFix/v2019-12/GMI/2x2.5/surface_VMRs_GMI_1959.2x25.nc`

Sorry it may be too messy here to pass all information. To be specific, here is the example containing all the problems I have stated above. I want to get all missing files for simulations of GCHP v13.2 from 2017-12-01 to 2018-12-31 with meteorology GEOS-FP and thus only high-resolution files are needed.
By running `bashdatacatalog-list -am -r 2017-12-01,2018-12-31 InputDataCatalogs/*.csv InputDataCatalogs/**/*.csv &> down-list-orig.txt` , I got the file attached below which contains files for year 2019 and 1959, as well as offline emission files with resolution of 0.5 degree:
[down-list-orig.txt](https://github.com/LiamBindle/bashdatacatalog/files/8894360/down-list-orig.txt)

The csv files I used are:
[EmissionsInputs.csv](https://github.com/LiamBindle/bashdatacatalog/files/8894363/EmissionsInputs.csv)
[MeteorologicalInputs.csv](https://github.com/LiamBindle/bashdatacatalog/files/8894364/MeteorologicalInputs.csv)
[ChemistryInputs.csv](https://github.com/LiamBindle/bashdatacatalog/files/8894365/ChemistryInputs.csv)

HEMCO_Config.rc is also attached here for reference:
[HEMCO_Config.rc.txt](https://github.com/LiamBindle/bashdatacatalog/files/8894385/HEMCO_Config.rc.txt)

**Comment by LiamBindle** [comment]
Hey Dandan. These are good questions.
1. This is correct. There is ~300 GB of data that the bashdatacatalog classifies as "always required". You need to download this data once, and after that the date range filtering will work as expected. 
  
   (Details if interested: The first and last year of climatologies need to be classified as _static assets_. This is so that "clamping" the first/last year of data to before/after the climaology period works.)
2. Emissions collections need to be enabled/disable manually. It would be great to automate this in the future, but for the time being it needs to be done manually. You should edit the emissions catalog file so it only downloads the 0.25deg offline emissions. By default the catalog files are configured for GCHP and GC-Classic with MERRA2.
3. This is correct. The reason is the same as (1). 

In summary:
* Catalog files need to be edited manually
* Your `down-list-orig.txt` looks correct
* The "extra" files you see in the list are files that the bashdatacatalog needs to classify as "always required"...they're so that climatologies work.

**Comment by 1Dandan** [comment]
Hi Liam, thanks for explanation. It is much clearer. I can understand downloading old years' data, but the data for 2019, especially for months after January should be unnecessary? 

For emissions, I tried to change the entry for offline emissions but it will add additional resolution path. For example, when I modify the line of offline dust in EmissionInputs.csv to:
`HEMCO/OFFLINE_DUST/v2021-08/0.25x0.3125` and corresponding url to: `http://geoschemdata.wustl.edu/ExtData/HEMCO/OFFLINE_DUST/v2021-08/0.25x0.3125`
It will still add resolution to the path and thus when I run the same command, the url becomes:
`http://geoschemdata.wustl.edu/ExtData/HEMCO/OFFLINE_DUST/v2021-08/0.25x0.3125/0.25x0.3125` which is not right.
How can I point the path to only certain directory?

**Comment by LiamBindle** [comment]
Hi Dandan. 

> I can understand downloading old years' data, but the data for 2019, especially for months after January should be unnecessary?

This is because of (1) in my response above. For the offline dust collection, the first and last year of data that is available is 2014/2019. This means 2014 and 2019 need to be classified as "always" because simulations before 2014 need the 2014 files, and simulations after 2019 need the 2019 files. 

> For emissions, I tried to change the entry for offline emissions but it will add additional resolution path... How can I point the path to only certain directory?

The offline dust collection is a single collection, so there isn't a way to exclusively get the 0.25deg files. Instead, you'll always get the 0.25deg and 0.5deg files. This means you will download some extra files that aren't needed, but the benefit is that configuring the catalogs is easier. I didn't want to make the emissions collections too fine-grain or else configuring `EmissionsInputs.csv` would be too painful.

Hope that helps. Let me know if you have any other questions.

**Comment by 1Dandan** [comment]
Hi Liam. Thanks, it makes sense now.
